Add GUI

cid0rz opened the pull request:
Hello, I developed a little GUI for begoneads, in case you want to add it to your repo :). I just slightly modified your scripts to be able to do what I needed for the GUI.

 I like a lot your program, thank you

cid0rz committed:
Add begoneads GUI

cid0rz committed:
made selection work as intended

cid0rz committed:
wip add record

cid0rz committed:
add record

cid0rz committed:
visual selection improvements

cid0rz committed:
small changes

cid0rz committed:
restructure of the code

cid0rz committed:
restructure and make code work with begoneads

cid0rz committed:
wip delete

cid0rz committed:
Delete working

cid0rz committed:
rearrangement

cid0rz committed:
copyright

cid0rz committed:
if name = main

cid0rz committed:
make a backups section

ShaunKulesa committed:
Create backup.txt

cid0rz committed:
create menu for backups

cid0rz committed:
Merge remote-tracking branch 'origin/master'

cid0rz committed:
wip back ups

cid0rz committed:
wip

cid0rz committed:
Create backups working

cid0rz committed:
deleting works

cid0rz committed:
timestamp converted to readable format

cid0rz committed:
fixed deleting

cid0rz committed:
fix edge cases

cid0rz committed:
restore of backup working

cid0rz committed:
Allow for pausing

cid0rz committed:
make check possible in command line and GUI

cid0rz committed:
update readme

cid0rz committed:
wip readme

cid0rz committed:
Update README.md

cid0rz committed:
fix readme

cid0rz committed:
Merge remote-tracking branch 'origin/master'

cid0rz committed:
Update README.md

cid0rz committed:
minor improvements

cid0rz committed:
typo

cid0rz committed:
add a desktop shoprtcut for the GUI in KDE

cid0rz committed:
Revert "Create backup.txt"

This reverts commit 6f78a77be686d15e33cd017ac666e12f8d533a98.

anned20 commented on begoneads.desktop:
```diff
@@ -0,0 +1,5 @@
+[Desktop Entry]
+Type=Application
+Name=Begoneads
+Exec=konsole --nofork -e sudo python /home/cid0rz/programs/begoneads/begoneadsGUI.py
```
Not everyone has konsole, nor is everyone's username cid0rz. This needs to be dynamic.

cid0rz replied:
Im working on this, The problem I am having is I am trying to move begoneadsGUI.py inside 'begoneads' subfolder, so it gets installed together with the other .py files of the package and then I am working a way of automatically detect (atleast in linux) where pip istalled begoneads, and fetch the GUI path from there. The imports get weird if i move the begoneadsGUI inside the other folder, Im trying to find a solution. Also for windows i think i can prepare a shortcut too. I havent tried the GUI on windows, I will try to test it. I have also some ideas for furhter improvements :)

anned20 replied:
I'll await your updates :)

anned20 commented on begoneads/begoneads.py:
```diff
@@ -135,6 +135,32 @@ def uninstall():
+        path = r'c:\windows\system32\drivers\etc\hosts'
+    else:
+        path = '/etc/hosts'
+        print("\nhost Linux")
```
This print is only for debugging purposes I think, it is not needed here.

anned20 commented on begoneads/begoneads.py:
```diff
@@ -135,6 +135,32 @@ def uninstall():
+        print("\nhost Linux")
+
+    hosts_manager = HostsManager(path)
+    chk= hosts_manager.has_begoneads()
```
```suggestion
    has_begoneads = hosts_manager.has_begoneads()
```

anned20 commented on begoneads/begoneads.py:
```diff
@@ -135,6 +135,32 @@ def uninstall():
+
+    hosts_manager = HostsManager(path)
+    chk= hosts_manager.has_begoneads()
+    if not chk:
```
```suggestion
    if not has_begoneads:
```

anned20 commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# BeGoneAds
+# BeGoneAds (fork with GUI)
```
This should be reverted back to the original title to merge this PR.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+            
+    def on_right_click(self, event):
+        item = self.tw.identify('item', event.x, event.y)
+        #print(item)
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+            self.tw.item(str(item), tags=())
+        else:
+            self.tw.item(str(item), tags='todelete')
+        #print("tag has: ", self.tw.tag_has('todelete'))
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+        self.remote_section.grid(padx=10, pady=20, row=0, column=0)
+        self.sources = list(enumerate(bg.sources))
+        self.local_sources = []
+        #for i in self.sources:
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+        sel2=self.local_section.tw.selection()
+        remotes = ",".join([self.remote_section.tw.item(i)['values'][1] for i in sel])
+        paths = ",".join([self.local_section.tw.item(i)['values'][1] for i in sel2])
+        #print (remotes)
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+        remotes = ",".join([self.remote_section.tw.item(i)['values'][1] for i in sel])
+        paths = ",".join([self.local_section.tw.item(i)['values'][1] for i in sel2])
+        #print (remotes)
+        #print(paths)
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+        paths = ",".join([self.local_section.tw.item(i)['values'][1] for i in sel2])
+        #print (remotes)
+        #print(paths)
+        #print([i.strip() for i in remotes.split(',')])
```
This can be removed.

anned20 commented on begoneadsGUI.py:
```diff
@@ -0,0 +1,321 @@
+        
+
+
+#if __name__ == __main__:        
```
This should be used.

anned20 commented:
Looks good, but needs a few touch ups.

Please try to be consistent with whitespace, for example; in the `begoneadsGUI.py` file, the amount of empty rules between functions varies.

cid0rz committed:
Apply suggestions from code review

Co-authored-by: Anne Douwe Bouma <annedouwe@bouma.tech>

cid0rz committed:
apply suggestions from review

cid0rz committed:
rearrange imports

cid0rz commented:
I have updated the imports so the gui program can go inside the same folder as the others. I have tested it on linux, i didnt have time to test on windows. If you want I can check tomorrow. Cheers!

cid0rz commented:
I checked on windows and it works very well. You need to run it from elevated cmd prompt. I am finishing the readme review and identation coherence. I re-requested review too soon :D sorry

cid0rz committed:
adapt readme and code style for merge

cid0rz commented:
I htink it is ready for your review. Thx

cid0rz commented:
Do you need any more changes? :)

anned20 commented on README.md:
```diff
@@ -9,7 +9,11 @@
 BeGoneAds is a script that puts some popular hosts file lists into the systems hosts file as a adblocker measure.
 
-See it working on asciinema:
+It has a command line and a GUI to help manage the hosts file. The intereface allows for installing Begoneads (play), uninstalling Begoneads (stop) and pausing begoneads, that creates a backup of the hosts file and uninstalls it. With the GUI you can easily select the sources you are going to install in your hosts file and you can also make and restore backups which allows for quick hosts file sw ...
```
```suggestion
It has a command line and a GUI to help manage the hosts file. The interface allows for installing BeGoneAds (play), uninstalling BeGoneAds (stop) and pausing BeGoneAds, that creates a backup of the hosts file and uninstalls it. With the GUI you can easily select the sources you are going to install in your hosts file and you can also make and restore backups which allows for quick hosts file switching.
```

anned20 commented on begoneads/begoneads.desktop:
```diff
@@ -0,0 +1,5 @@
+[Desktop Entry]
+Type=Application
+Name=Begoneads
+Exec=xterm -e sudo python $(find $(pip show begoneads | grep Location | awk '{print $2}') -name "begoneadsGUI.py")
```
What happens when the user does not have `xterm` installed?

The `find` construction also doesn't look very durable. Is there a better solution to this?

cid0rz replied:
idk about this. you need something to run the script in. xterm should be installed in most linux operating systems. this is a convenience you can also run the begoneadsGUI script and it will work

anned20 commented on begoneads/begoneads.py:
```diff
@@ -136,5 +136,27 @@ def uninstall():
+    # Check if we have sufficient permissions
+    if not is_admin(sys.platform.startswith('win')):
+        raise NotElevatedException(
+            'This program needs to be run as root to work properly')
```
```suggestion
            'This program needs to be run as root to work properly'
        )
```

anned20 commented on begoneads/begoneadsGUI.py:
```diff
@@ -0,0 +1,322 @@
+# Copyright  (c) 2022 Cid0rz (cid.kampeador@gmail.com)
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
```
This is not needed, the license is not included in the source files in this repository.

cid0rz committed:
Update README.md

Co-authored-by: Anne Douwe Bouma <annedouwe@bouma.tech>

cid0rz committed:
Update begoneads/begoneads.py

Co-authored-by: Anne Douwe Bouma <annedouwe@bouma.tech>

snyk-bot committed:
fix: requirements.txt to reduce vulnerabilities


The following vulnerabilities are fixed by pinning transitive dependencies:
- https://snyk.io/vuln/SNYK-PYTHON-REQUESTS-5595532

cid0rz committed:
Merge pull request #1 from cid0rz/snyk-fix-2e5237686246bfd42485bcac7e0d1e17

[Snyk] Security upgrade requests from 2.28.1 to 2.31.0
